### PR TITLE
Resources tutors

### PR DESCRIPTION
**Pull request by BoazFrancis**
Create tutor resources page and show resources.
Add upload new files to storage.

### CODE REVIEW

File: app/Http/Controllers/ResourceController.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+class ResourceController extends Controller
+{
+  public function index() 
+  {
+    if (Auth::user()->isTutor()) {
+       return view('resources');
+    }
+    else {
+      return;
```
Comment (jasonlipo) [inline app/Http/Controllers/ResourceController.php]: `abort(404)`?

File: app/Http/Controllers/ResourceController.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+      $resource = new Resource;
+      $resource->name = $request->filename;
+      $resource->move(public_path().'/tmp_resources', $resource->name);
+      $resource->save();
+
+      return back();
+    }
+  }*/
+
```
Comment (jasonlipo) [inline app/Http/Controllers/ResourceController.php]: Remove commented code above ^^^

File: app/StudentAssignment.php [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+class StudentAssignment extends Model
+{
+    public function student() {
+        return $this->belongsTo('App\Student', 'student_id', 'user_id');
+    }
+    public function tutor() {
+        return $this->belongsTo('App\Tutor', 'tutor_id', 'user_id');
+    }
+}
```
Comment (jasonlipo) [inline app/StudentAssignment.php]: 2 spaces please

File: app/Http/Controllers/ResourceController.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  {
+    if ($request->hasfile('filename'))
+    {
+      $file = $request->filename;
+      $name = $file->getClientOriginalName();
+      Storage::disk('s3')->put('resources/', $file, 'public');
+      return back();
+    }
+  }
+/*
```
Comment (jasonlipo) [inline app/Http/Controllers/ResourceController.php]: Tests for `ResourceController:list` and `ResourceController::store`

File: database/migrations/2018_06_04_175124_add_tutor_id_to_resources_table.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('resources', function (Blueprint $table) {
+            $table->integer('tutor_id')->unsigned();
+            $table->foreign('tutor_id')->references('user_id')->on('tutors');
+        });
+    }
```
Comment (jasonlipo) [inline database/migrations/2018_06_04_175124_add_tutor_id_to_resources_table.php]: 2 spaces please

File: database/migrations/2018_06_05_141111_add_student_id_to_student_assignment_table.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('student_assignments', function (Blueprint $table) {
+            $table->integer('student_id')->unsigned();
+            $table->foreign('student_id')->references('user_id')->on('students');
+        });
+    }
+
```
Comment (jasonlipo) [inline database/migrations/2018_06_05_141111_add_student_id_to_student_assignment_table.php]: 2 spaces above

**Comment by BoazFrancis** [comment]
Fixed issues, ready for review

**Pull request merged**